### PR TITLE
added a Dockerfile and made small changes in Makefile

### PR DESCRIPTION
**Pull request by ALimaRedHat**
Working with @rogbas on ticket https://jira.coreos.com/browse/SREP-1271

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -35,9 +35,14 @@ generate-oauth-templates:
 
 .PHONY: generate-syncset
 generate-syncset: generate-oauth-templates
-	docker run --rm -v `pwd`:`pwd` python:2.7.15 /bin/sh -c "cd `pwd`; pip install pyyaml; scripts/generate_syncset.py -t ${SELECTOR_SYNC_SET_TEMPLATE_DIR} -y ${YAML_DIRECTORY} -d ${SELECTOR_SYNC_SET_DESTINATION} -r ${REPO_NAME}"
+	if [ -z ${IN_DOCKER_CONTAINER} ]; then \
+		docker run --rm -v `pwd`:`pwd` python:2.7.15 /bin/sh -c "cd `pwd`; pip install pyyaml; scripts/generate_syncset.py -t ${SELECTOR_SYNC_SET_TEMPLATE_DIR} -y ${YAML_DIRECTORY} -d ${SELECTOR_SYNC_SET_DESTINATION} -r ${REPO_NAME}"; \
+	else \
+		pip install pyyaml; \
```
Comment (jewzaam) [inline Makefile]: I don't want `pip` installing anything in the script.  Make it a dependency, documented in README, and fail if it's not there.

File: Makefile [hunk 2]
```diff
@@ -35,9 +35,14 @@ generate-oauth-templates:
...
+		scripts/generate_syncset.py -t ${SELECTOR_SYNC_SET_TEMPLATE_DIR} -y ${YAML_DIRECTORY} -d ${SELECTOR_SYNC_SET_DESTINATION} -r ${REPO_NAME}; \
```
Comment (jewzaam) [inline Makefile]: Can the command to execute be moved to a single place (var) so it's no duplicated?

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+COPY . ${REPO_PATH}
+WORKDIR ${REPO_PATH}
+
+# Docker dependencies:
+RUN yum update -y
+RUN yum install -y git
+RUN yum install -y make
+
+# Makefile dependencies:
+RUN curl "https://bootstrap.pypa.io/get-pip.py" -o "get-pip.py"
```
Comment (rogbas) [inline Dockerfile]: this method of installing pip is a potential problem. If someday this url is off, our CI breaks as well and we can't merge PRs into that repo.

Comment (ALimaRedHat) [reply Dockerfile]: The preferred way is actually not working...
```
[root@a8652c0b01db /]# yum install python-pip python-wheel
Loaded plugins: fastestmirror, ovl
Determining fastest mirrors
 * base: mirrors.mit.edu
 * extras: mirrors.mit.edu
 * updates: mirrors.mit.edu
base                                                                                                                                                                    | 3.6 kB  00:00:00     
cbs-paas7-openshift-multiarch-el7-build                                                                                                                                 | 3.4 kB  00:00:00     
extras                                                                                                                                                                  | 3.4 kB  00:00:00     
updates                                                                                                                                                                 | 3.4 kB  00:00:00     
(1/6): base/7/x86_64/group_gz                                                                                                                                           | 166 kB  00:00:00     
(2/6): extras/7/x86_64/primary_db                                                                                                                                       | 205 kB  00:00:00     
(3/6): updates/7/x86_64/primary_db                                                                                                                                      | 6.5 MB  00:00:00     
(4/6): cbs-paas7-openshift-multiarch-el7-build/x86_64/group_gz                                                                                                          |  458 B  00:00:00     
(5/6): base/7/x86_64/primary_db                                                                                                                                         | 6.0 MB  00:00:00     
(6/6): cbs-paas7-openshift-multiarch-el7-build/x86_64/primary_db                                                                                                        | 6.2 MB  00:00:02     
No package python-pip available.
No package python-wheel available.
Error: Nothing to do
```

File: Makefile [hunk 4]
```diff
@@ -35,9 +38,13 @@ generate-oauth-templates:
 
 .PHONY: generate-syncset
 generate-syncset: generate-oauth-templates
-	docker run --rm -v `pwd`:`pwd` python:2.7.15 /bin/sh -c "cd `pwd`; pip install pyyaml; scripts/generate_syncset.py -t ${SELECTOR_SYNC_SET_TEMPLATE_DIR} -y ${YAML_DIRECTORY} -d ${SELECTOR_SYNC_SET_DESTINATION} -r ${REPO_NAME}"
+	if [ -z ${IN_DOCKER_CONTAINER} ]; then \
+		docker run --rm -v `pwd`:`pwd` python:2.7.15 /bin/sh -c "cd `pwd`; pip install pyyaml; ${GEN_SYNCSET}"; \
```
Comment (lisa) [inline Makefile]:
```suggestion
		docker run --rm -v `pwd -P`:`pwd -P` python:2.7.15 /bin/sh -c "cd `pwd`; pip install pyyaml; ${GEN_SYNCSET}"; \
```

Support for symlinks

Comment (ALimaRedHat) [reply Makefile]: Done

File: Makefile (hunk 2 above)
Comment (ALimaRedHat) [reply Makefile]: Done

File: Makefile (hunk 1 above)
Comment (ALimaRedHat) [reply Makefile]: Done

**Comment by lisa** [comment]
What is the Dockerfile's purpose here? I don't see a reference to it with a `docker build`?

**Comment by rogbas** [comment]
> What is the Dockerfile's purpose here? I don't see a reference to it with a `docker build`?

The goal is linking this repo with `openshift/release`, that requires the component to provide a Dockerfile to be built by prow for CI/testing purposes.

**Comment by lisa** [comment]
> > What is the Dockerfile's purpose here? I don't see a reference to it with a `docker build`?
> 
> The goal is linking this repo with `openshift/release`, that requires the component to provide a Dockerfile to be built by prow for CI/testing purposes.

Ohh, I see. Is there a PR in that repo that we can reference here? (or vice versa)

**Comment by ALimaRedHat** [comment]
@lisa this is the PR in `openshift/release`: https://github.com/openshift/release/pull/4323

**Pull request merged**